Zend/zend_execute.c

```diff
@@ -3055,6 +3055,27 @@ static zend_never_inline bool ZEND_FASTCALL zend_array_key_exists_fast(HashTable
 	}
 }
 
+static zend_never_inline bool ZEND_FASTCALL zend_array_key_exists_for_array_access(zend_object *obj, zval *key OPLINE_DC EXECUTE_DATA_DC)
+{
+	if (!instanceof_function(obj->ce, zend_ce_arrayaccess)) {
```

Comment:
This condition is redundant, maybe replace it with an `assert`.

Reply:
Oh, how I didn't notice it, thanks for spotting it! It is copy-pasted from array.c, so I guess it would be worth to remove the code duplication.

Reply:
BTW I don't understand why the `zend_never_inline` is added to `zend_array_key_exists_fast`. Doesn't it make the code slower?

Reply:
@kocsismate To avoid it from being inlined in the various `ZEND_ARRAY_KEY_EXISTS` variations. The same makes sense in your case.